src/netutil.rs

```diff
@@ -49,21 +49,42 @@ pub fn parse_port_or_addr(optstr: Option<&str>, default_port: u16) -> SocketAddr
 }
 
 pub fn get_ip_addr() -> Option<IpAddr> {
-    for iface in datalink::interfaces() {
+    let mut ifaces = datalink::interfaces();
+
+    // put eth0 and wifi0, etc. up front of our list of candidates
```

Comment:
Can you move everything from here on down into a separate function and add tests for it?

Reply:
sure